ibm/service/vpc/resource_ibm_is_vpc.go

```diff
@@ -1170,3 +1389,290 @@ func suppressNullAddPrefix(k, old, new string, d *schema.ResourceData) bool {
+		if err != nil {
+			return model, err
+		}
+		model.Resolver = ResolverModel
+	}
+	return model, nil
+}
+
+func resourceIBMIsVPCMapToVpcdnsResolverPrototype(modelMap map[string]interface{}) (vpcv1.VpcdnsResolverPrototypeIntf, error) {
+	model := &vpcv1.VpcdnsResolverPrototype{}
+	if modelMap["type"] != nil && modelMap["type"].(string) != "" {
+		model.Type = core.StringPtr(modelMap["type"].(string))
+	}
+	if modelMap["manual_servers"] != nil {
+		model.Type = core.StringPtr("manual")
+		manualServers := []vpcv1.DnsServerPrototype{}
+		for _, manualServersItem := range modelMap["manual_servers"].([]interface{}) {
+			manualServersItemModel, err := resourceIBMIsVPCMapToDnsServerPrototype(manualServersItem.(map[string]interface{}))
+			if err != nil {
+				return model, err
+			}
+			manualServers = append(manualServers, *manualServersItemModel)
+		}
+		model.ManualServers = manualServers
+	}
+	return model, nil
+}
+
+func resourceIBMIsVPCMapToDnsServerPrototype(modelMap map[string]interface{}) (*vpcv1.DnsServerPrototype, error) {
+	model := &vpcv1.DnsServerPrototype{}
+	model.Address = core.StringPtr(modelMap["address"].(string))
+	if modelMap[isVPCDnsResolverManualServersZoneAffinity] != nil && modelMap[isVPCDnsResolverManualServersZoneAffinity].(string) != "" {
+		ZoneAffinityModel := &vpcv1.ZoneIdentity{
+			Name: core.StringPtr(modelMap[isVPCDnsResolverManualServersZoneAffinity].(string)),
+		}
+		model.ZoneAffinity = ZoneAffinityModel
+	}
+	return model, nil
+}
+
+// for dns patch
+
+func resourceIBMIsVPCMapToVpcdnsPatch(modelMap map[string]interface{}) (*vpcv1.VpcdnsPatch, error) {
+	model := &vpcv1.VpcdnsPatch{}
+	if modelMap["enable_hub"] != nil {
+		model.EnableHub = core.BoolPtr(modelMap["enable_hub"].(bool))
+	}
+	if modelMap["resolver"] != nil && len(modelMap["resolver"].([]interface{})) > 0 {
+		ResolverModel, err := resourceIBMIsVPCMapToVpcdnsResolverPatch(modelMap["resolver"].([]interface{})[0].(map[string]interface{}))
+		if err != nil {
+			return model, err
+		}
+		model.Resolver = ResolverModel
+	}
+	return model, nil
+}
+
+func resourceIBMIsVPCMapToVpcdnsResolverPatch(modelMap map[string]interface{}) (*vpcv1.VpcdnsResolverPatch, error) {
+	model := &vpcv1.VpcdnsResolverPatch{}
+	if modelMap["manual_servers"] != nil {
+		manualServers := []vpcv1.DnsServerPrototype{}
+		for _, manualServersItem := range modelMap["manual_servers"].([]interface{}) {
+			manualServersItemModel, err := resourceIBMIsVPCMapToDnsServerPrototype(manualServersItem.(map[string]interface{}))
+			if err != nil {
+				return model, err
+			}
+			manualServers = append(manualServers, *manualServersItemModel)
+		}
+		model.ManualServers = manualServers
+	}
+	if modelMap["type"] != nil && modelMap["type"].(string) != "" {
+		model.Type = core.StringPtr(modelMap["type"].(string))
+	}
+	if modelMap["vpc"] != nil && len(modelMap["vpc"].([]interface{})) > 0 {
```

Comment:
Handle this scenario 
"Specify null to remove an existing VPC."
https://pages.github.ibm.com/vpc/vpc-spec-artifacts/branch/master/swagger-ui.html?feature=is-vpc-dns-sharing-topology-for-endpoint-gateways#/VPCs/update_vpc

Reply:
addressed